### DISCUSSION TITLE
Customizing colors in Windows

### DISCUSSION THREAD

**Discussion by Inventhrice** [lsd-rs/lsd#958] [Q&A] [answered]
Hello!

I have already seen this [discussion](https://github.com/orgs/lsd-rs/discussions/898#discussion-5603900), and wish to follow up on it. I am currently running lsd on Windows, and I saw that colors.yaml is not implemented yet. Until that feature is implemented, what method are Windows users supposed to use to customize the directory colors?

Thanks!

**Comment by Inventhrice** [comment]
Full disclosure, I've been trying to look through the code and the documentation to find an answer, and could not find it haha! I don't believe `$LS_COLORS` is an environment variable that exists on Windows, and all the instructions seems to be geared towards Linux, so I am a bit lost.

**Reply by zwpaper** [reply]
Hi @Inventhrice, we use the std env to get `LS_COLORS`, it should also work on Windows, maybe you should give it a try

**Reply by Inventhrice** [reply]
Hi @zwpaper ! Thank you for your response! I've given it a shot, however, nothing really seems to work. I'm not too familiar with stv env's in Windows, so I tried to figure it out (hence I might've done it wrong). I've attached a screenshot below, if you have any further advice that would be much appriciated!

![image](https://github.com/lsd-rs/lsd/assets/45127310/ec280053-a492-401e-962d-a37c1109e24a)

Cheers!

**Reply by zwpaper** [reply]
try setx?

```sh
setx LS_COLORS "xxxxxx"
```

**Reply by Inventhrice** [reply]
Unfortunately no luck there either.
![image](https://github.com/lsd-rs/lsd/assets/45127310/7b47fee6-aeca-407a-b456-ff3438b78161)

![image](https://github.com/lsd-rs/lsd/assets/45127310/f6dccd14-dfdf-4a08-925b-a2116c7c06f0)

(And I just wanted to post confirmation that I am using the correct settings in config.yaml)
![image](https://github.com/lsd-rs/lsd/assets/45127310/f30a3a9e-34b1-4086-827c-33b6e5907c1f)

**Reply by Inventhrice** [reply]
Hi! Followup:
It worked! I'm not sure what I did, and I think I might've found a bug?

![image](https://github.com/lsd-rs/lsd/assets/45127310/40170be7-505d-4cc8-82d5-caedac924acd)

It changed it to the appropriate setting! But it seems like it only occurs on reboots, because after I saw this I decided to try another setting and it reverted to default settings.

**Reply by Inventhrice** [reply]
I decided to test this, and this is what I've found:
- using SET LS_COLORS does not work
- SETX LS_COLORS "[value]" **does** work, after a restart.

If you know any way to force lsd to use the new LS_COLORS value (either by reloading the program somehow maybe? or some other wizardary), that'd be greatly appriciated!

EDIT: I have found the answer, and I am detailing it in a seperate comment.

**Comment by Inventhrice** [comment, marked as answer]
A temporary solution is as follows (thank you @zwpaper for your help!)

1. `setx LS_COLORS "[value]"` (example `setx LS_COLORS "di=35"`)
2.  `RefreshEnv`
3.  Done!

To find what values can be put into LS_COLORS, refer to [this website](https://geoff.greer.fm/lscolors/).

NOTE: RefreshEnv is a command that comes in with Chocolatey. The reason the changes do not take effect immediately is because apparently Windows loads a temp copy of the system variables and uses that. The script that chocolatey uses ([found here](https://github.com/chocolatey/choco/blob/develop/src/chocolatey.resources/redirects/RefreshEnv.cmd)) only takes effect for the command line session ONLY.

To make the changes permanent, restart your machine after you have made all the necessary changes. The above commands are only to facilitate making sure that you like your configuration before restarting.
